standalone/auth-wrapper.jsx

```diff
@@ -85,10 +85,21 @@ export const AuthWrapper = Wrapped => class extends Component {
         window.location.replace(uri);
     };
 
+    setTracksUser = () =>
+        this.state.wpcom
+        .me()
+        .get( { fields: 'ID,username' } )
+        .then( ( { ID, username } ) => {
+            window._tkq = window._tkq || [];
+            window._tkq.push( [ 'identifyUser', ID, username ] ) ;
+        } )
+        .catch( () => {} );
```

Comment:
on failure I think we should no-op any tracks requests, otherwise we'll still run into the current situation.

Reply:
I'll add a "do not set cookie on `wp.com`" as a last mile safeguard in `w.js` which means you will get unreliable IDs across page reloads ( demultiplying `calypso_notifications_*` uniques ) if we continue to send pixels without an `identifyUser`, but we won't set cookies on `wp.com` no matter what. It's up to you folks to decide between: a bit more data ( raw counts ) with slightly unreliable uniques, and less data with 100% reliable uniques.

Reply:
Thanks @bperson we'll try and aim for more data with less reliable uniques. Let us know when the w.js update is ready and we can bump the cache break.

Reply:
> on failure I think we should no-op any tracks requests, otherwise we'll still run into the current situation.

> It's up to you folks to decide between: a bit more data ( raw counts ) with slightly unreliable uniques, and less data with 100% reliable uniques.

I'd agree with @gwwar , if we don't have a valid authenticated user for some reason, we should just drop making the request. However, I don't see how we can tell if tracks considers the user known or not, _before_ making the `window._tkq.push()` call. We can't rely on the presence, absence, nor value of a `tk_ai` cookie, since it may not have been established yet (a successful `identifyUser` call just makes sure that we _don't_ have one). We could add something to the app's state upon the successful `me()` call (or use the presence of an existing `this.state.oAuthToken` to assume it's all good), but it would be nice to ask Tracks directly whether Tracks thinks the user is known or not.

@bperson could we either 1) have `identifyUser` be more helpful in how it handles cookies, or 2) add something to TKQ's API that we could access, eg. `window._tkq.userAuthenticated()`? Or maybe I'm missing an obvious easier solution?

Reply:
@gwwar @kwight  If you're going for more data but less reliable uniques, you should just noop on error when fetching `/me`: Tracks will not use the logged-in user ID ( because ... we couldn't get one ;) ), create an anonymous ID at the first `recordEvent`, try to back it by a cookie which will fail because of the ( upcoming ) safeguard against `wp.com` in `w.js`. On subsequent reload ( or visiting any other site ), we won't be able to use the same anonymous ID because it wasn't persisted, so any new event will be associated with a completely different ID, leading to more unreliable uniques.

So if you're going with this solution, I think this PR looks good and there is nothing else to do on your end, except bumping the cache break in a couple of hours when I get the updated `w.js` out. 

Just for completing the thought process, if you were to go for less data with more reliable uniques, we would have to "block" sending `recordEvent` if we aren't able to `identifyUser`, the best way to accomplish that without encountering the state issue you mentioned @kwight ( not knowing if we've loaded an ID from `/me` when we push a `recordEvent` for instance ), is to let it push, but never load `w.js` to never send a pixel. And the easiest way to do that would be to load `w.js` on fetch success for `/me` instead of in the default template of the iframe content I believe.